lib/SILOptimizer/Mandatory/SemanticARCOpts.cpp

```diff
@@ -251,6 +247,23 @@ bool SemanticARCOptVisitor::processWorklist() {
     }
   }
 
+  // Then eliminate the rest of the dead trivial insts.
+  //
+  // NOTE: We do not need to touch the worklist here since it is guaranteed to
+  // be empty due to the loop above. We enforce this programatically with the
+  // assert.
+  assert(worklist.empty() && "Expected drained worklist so we don't have to "
+                             "remove dead insts form it");
+  while (!deadTrivialInsts.empty()) {
+    auto val = deadTrivialInsts.pop_back_val();
+    if (!val)
+      continue;
+    recursivelyDeleteTriviallyDeadInstructions(
+        *val, true /*force*/,
```

Comment:
Out of curiosity, why is `force=true` appropriate here?  Also, why was it previously necessary to add operands and results to the worklist but is no longer?

Reply:
The reason why I put force here is b/c if I don't put force it just does the isTriviallyDead check again.

The reason why I removed the operand/result thing is I find with the test case in question that recursivelyDeleteTriviallyDeadInstructions can re-add the struct_extract from the test case to the worklist as a result of an instruction.

I just avoided all of the problems by doing it later.

Reply:
> The reason why I put force here is b/c if I don't put force it just does the isTriviallyDead check again.

Anytime we violate the spirit of the API like this there should be a comment explaining the cleverness.